fms_extras/utils/generation.py

```diff
@@ -106,26 +435,19 @@ def speculative_generate(
         for line, input_id in zip(result, list(input_ids))
     ]
     block_mapping_max = ((max_len + new_tokens) // 16) + 1
```

Comment:
Do we want to make this `+2` instead of `+1`, given that the speculator may produce up to `n_predict-1` extra tokens at the final step?

Reply:
yes this makes sense,

```
(max_len + new_tokens + inp_len - 1) // 16 )
```

I'm not sure this is right though:

max_len=48
new_tokens=10
inp_len=4

61 // 16 => 3 -> We really would want to have 4 for this since there are tokens left over.

I think the way we need to assure this works is the following:

```python
((max_len - new_tokens + input_len - 1) // 16) + 1
```

Reply:
Added in latest commit